docs/source/en/data/task/index.md

```diff
@@ -60,17 +60,17 @@ Callbacks aren't very composable. In order to combine callbacks, an user has to
       $ASSERT(v == null);
     });
 
-Because no function using callbacks ever returns a value to the caller, and so aren't usable with most functions. They are, of course, not usable with JavaScript control-flow constructs either, so it's not possible to write something like:
+Because functions using callbacks don't return a value to the caller, they're not composable. They are also, of course, not usable with JavaScript control-flow constructs either. So it's not possible to write something like:
 
     if (someAsyncPredicate(...)) {
       ...
     }
 
-Since `someAsyncPredicate` never returns any value, but instead passes it as an argument to another function.
+Since `someAsyncPredicate`, instead of returning a value, passes the result of its computation to another function (the callback).
```

Comment:
Come to think of it: 

"Instead of returning a value, `someAsyncPredicate` passes the result..."

would flow better. Feel free to change this in your branch if you want to :)